http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -90,6 +109,8 @@ public void messageReceived(final String message) {
 
   @Override
   public void connectionClosed(final String reason) {
+    connectionClosedListeners.forEach(
+        connectionLostListener -> connectionLostListener.accept(reason));
     connectionLostListeners.forEach(
         connectionLostListener -> connectionLostListener.accept(reason));
```

Comment:
Ok, what exactly is the difference between those 2 listeners?

Reply:
Please let me know if these javadocs don't quite answer that question:
```
  /** These are called if connection is disconnected (by server). */
  private final Collection<Consumer<String>> connectionLostListeners = new ArrayList<>();
  /** These are called whenever connection is closed, whether by us or server. */
  private final Collection<Consumer<String>> connectionClosedListeners = new ArrayList<>();
```

Reply:
Ah, sorry I should've been more clear. I get that this was the intended difference, but right now the only usage I see is in line 112, where the exact same thing is passed the connection lost listeners, the only difference being the order in which both are triggered.

So it currently looks like there is now difference from a code perspective.

Reply:
The fix here is maybe not quite as clean as it could be (ie: slightly hacky), notice in "close" we clear the connection lost listeners.

This means if we have a 'connectionClosed' by the underlying socket, we'll call both listeners. If we call 'close' ourselves, then the 'connectionLost' listener set will be empty and we'll still call the 'connectionClosedListeners'.

Reply:
Ahh, thanks for clarifying.
Maybe worth adding in a comment to explain this to future versions of ourselves?

Reply:
:grin:   7bbe6f9 
Done